Library/Homebrew/sandbox.rb

```diff
@@ -165,6 +162,9 @@ class SandboxProfile
           (regex #"^/dev/ttys?[0-9]*$")
           )
       (deny file-write*) ; deny non-whitelist file write operations
+      (deny file-read*
+          (regex #"^/Users/[a-zA-Z0-9_]+/.*_history$")
+          )
```

Comment:
Hey Xu, good to see you again. Thanks for the advice.

1 & 2 are perfectly fair points. I did actually have them explicitly written at first but I wanted to avoid introducing large clumps of code, reasoning the more I wrote the harder it would be to get through review 😅.

> I would not recommend to add this rule directly in SandboxProfile base template. Instead, can you create Sandbox#deny_read_shell_history method? This can also be helpful, say not enforcing such rule under --interactive mode.

My broader plan was to introduce a `deny_read` & related method, add shell history as well as user downloads/pictures/documents/etc whilst not blocking off the whole of `$HOME` entirely as obviously there are Homebrew users who live there. But I was wary about getting that through review. But you're right, I guess if I'm going to do it I might as well do it properly.